server/routes/handlers/projects.js

```diff
@@ -42,6 +43,78 @@ module.exports = {
     (req, res) => res.status(200).json(res.locals.project),
   ],
 
+  put: [
+    (req, res, next) => {
+      req.app.locals.users = req.body.users;
+      delete req.body.users;
+      return projectController
+        // Save by ensuring the payload contains the proper id which is used for perms
+        .update(req.app.locals.project, Object.assign({}, req.body, { id: req.params.id }))
+        // Pull the original list of users so we can diff in next step
+        // User_association helps us retrieve the users's roles in the project based on user_id 
+        .then(p => p.refresh({ withRelated: ['users', 'userAssociation'] }))
+        .then((project) => {
+          res.locals.project = project;
+          return next();
+        })
+        .catch((err) => {
+          req.app.locals.logger.error(err);
+          return next(new Error('Error while saving your project.'));
+        });
+    },
+    // Delete missing users
+    (req, res, next) => {
+      const origProject = res.locals.project.toJSON();
+      const origUsers = origProject.users;
+      const users = req.app.locals.users;
+      const missingUsers = projectController.getMissingUsers(
+        origUsers,
+        users,
+        origProject.userAssociation,
+      );
+      // Delete sync
+      return projectController.removeUsers(origProject.id, missingUsers)
+        .then(userController.removeUsers(missingUsers))
```

Comment:
Should be `.then(() => userController.removeUsers(missingUsers))`, otherwise this won't wait for the associations to be deleted, which could cause foreign key errors.